featurebyte/query_graph/node/count_dict.py

```diff
@@ -119,6 +174,118 @@ def generate_expression(self, operand: str, other_operands: List[str]) -> str:
+        return statements, extreme_value_key_var
+
+    @staticmethod
+    def _get_unique_count(
+        count_dict_var_name: str,
+        var_name_generator: VariableNameGenerator,
+        include_missing: bool,
+    ) -> Tuple[List[StatementT], VarNameExpressionInfo]:
+        if include_missing:
+            unique_count_expr = get_object_class_from_function_call(
+                f"{count_dict_var_name}.apply",
+                ExpressionStr("len"),
+            )
+        else:
+            unique_count_expr = get_object_class_from_function_call(
+                f"{count_dict_var_name}.apply",
+                ExpressionStr(
+                    f"lambda x: len([key for key in x if key != '{MISSING_VALUE_REPLACEMENT}'])"
+                ),
+            )
+
+        unique_count_var = var_name_generator.convert_to_variable_name(
+            variable_name_prefix="feat_uniq_cnt", node_name=None
+        )
+        statements: List[StatementT] = [(unique_count_var, unique_count_expr)]
+        return statements, unique_count_var
+
+    def _derive_on_demand_view_code(
+        self,
+        node_inputs: List[VarNameExpressionInfo],
+        var_name_generator: VariableNameGenerator,
+        config: OnDemandViewCodeGenConfig,
+    ) -> Tuple[List[StatementT], VarNameExpressionInfo]:
+        statements, cd_var_name, mask_var = self._get_count_dict_and_mask_variables(
+            node_inputs, var_name_generator
+        )
+        if self.parameters.transform_type == "entropy":
+            entropy_statements, entropy_var = self._get_entropy(cd_var_name, var_name_generator)
+            statements.extend(entropy_statements)
+            return statements, ExpressionStr(f"{entropy_var}.reindex({mask_var}.index)")
+        if self.parameters.transform_type in {"most_frequent", "key_with_highest_value"}:
+            most_freq_statements, most_freq_var = self._get_extreme_value_key(
+                cd_var_name, var_name_generator, operation="max"
+            )
+            statements.extend(most_freq_statements)
+            return statements, ExpressionStr(f"{most_freq_var}.reindex({mask_var}.index)")
+        if self.parameters.transform_type in {"key_with_lowest_value"}:
+            least_freq_statements, least_freq_var = self._get_extreme_value_key(
+                cd_var_name, var_name_generator, operation="min"
+            )
+            statements.extend(least_freq_statements)
+            return statements, ExpressionStr(f"{least_freq_var}.reindex({mask_var}.index)")
+        if isinstance(self.parameters, self.UniqueCountParameters):
+            unique_count_statements, unique_count_var = self._get_unique_count(
+                cd_var_name, var_name_generator, self.parameters.include_missing
+            )
+            statements.extend(unique_count_statements)
+            return statements, ExpressionStr(f"{unique_count_var}.reindex({mask_var}.index)")
```

Comment:
The code here seems a bit dense but there are some repetitions: the only difference is the method that gets called. Would be good if it can be refactored.